### PR TITLE
Rewrite functions too

### PR DESCRIPTION
**Pull request by ionathanch**
Might as well, right?
Gotta change the writeup to include this though, but functions are similar enough to methods that it can probably just be mentioned in passing. The only difference is that the body of a function is an expression, not a statement.

### CODE REVIEW

File: src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala [hunk 1]
```diff
@@ -49,26 +53,28 @@ trait InlineRewrite extends PredicateExpansion {
...
     * @param stmts A Seqn whose statements will be traversed.
-    * @param method The method containing the inhale statement we wish to expand.
+    * @param method The member containing the inhale statement we wish to expand.
     * @param program The Viper program for which we perform this expansion.
     * @param cond The condition a predicate must satisfy to be expanded within an inhale statement.
     * @return The Seqn with all inhale, exhale, assert, and while loop statements expanded.
     */
-  private[this] def expandStatements(stmts: Seqn, method: Method, program: Program, cond: String => Boolean): Seqn =
+  private[this] def expandStatements(stmts: Seqn, member: Member, program: Program, cond: String => Boolean): Seqn = {
+    val noFoldUnfoldStmts = removeFoldUnfolds(stmts, cond)
```
Comment (ionathanch) [inline src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala:63]: Just as whenever you expand expression you also want to get rid of `Unfolding`s, when you expand statements, you'd always want to get rid of `Fold` and `Unfold` as well.

**Comment by jyoo980** [comment]
@ionathanch I can change the parts I've written about functions

**Review by jyoo980 — APPROVED** [review]
Looks great. I can think of a way we can remove the need for the specific `rewriteFunction` and `rewriteMethod` functions, but that seems like it's more trouble than it's worth + might be me trying to overengineer things. Thanks!

**Review by alison-li — APPROVED** [review]

**Comment by ionathanch** [comment]
I was thinking the same thing, since they're both `Contracted`, but yeah it seemed like it would make the code more complex to deduplicate things

**Comment by jyoo980** [comment]
I was thinking we'd define a sealed trait where we only declare the things we care about, then have `Function` and `Method` implement it and override those values specifically. They'd  then have a `.toFunction` or `.toMethod` thing we'd call at the end after we do the rewrite but I'm just yak shaving here

Please don't implement this lol update comment

**Review by braxtonhall — APPROVED** [review]
Why do `Method` and `Function` redundantly define `body` - shouldn't that be in a supertype? Wouldn't it be trivial to remove the code duplication if that were the case?

This irks me but 🚢

**Comment by jyoo980** [comment]
Because the body of `Function` is an `Option[Exp]`, while the body of `Method` is an `Option[Seqn]` (statements)

**Comment by ionathanch** [comment]
Just realized I haven't written any test examples, gonna do some (or one, at least) and merge after that

**Comment by jyoo980** [comment]
🅱️regor spinning in his grave rn

**Pull request merged**